core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/ImageFilterParamsFactory.java

```diff
@@ -0,0 +1,54 @@
+package com.sequenceiq.cloudbreak.service.upgrade;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.cloud.model.ClouderaManagerProduct;
+import com.sequenceiq.cloudbreak.cloud.model.catalog.Image;
+import com.sequenceiq.cloudbreak.cluster.service.ClouderaManagerProductsProvider;
+import com.sequenceiq.cloudbreak.common.exception.NotFoundException;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.domain.stack.cluster.ClusterComponent;
+import com.sequenceiq.cloudbreak.service.parcel.ParcelService;
+import com.sequenceiq.cloudbreak.service.upgrade.image.ImageFilterParams;
+
+@Component
+class ImageFilterParamsFactory {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ImageFilterParamsFactory.class);
+
+    @Inject
+    private ParcelService parcelService;
+
+    @Inject
+    private ClouderaManagerProductsProvider clouderaManagerProductsProvider;
+
+    ImageFilterParams create(Image image, boolean lockComponents, Stack stack) {
+        return new ImageFilterParams(image, lockComponents, getActivatedParcels(stack), stack.getType());
+    }
+
+    private Map<String, String> getActivatedParcels(Stack stack) {
```

Comment:
Probably we shouldn't call it getActivatedParcels anymore, because we're not checking the CM anymore. Perhaps something like `getStackRelatedParcels`.

Reply:
In the `ImageFilterParams` we should rename the `activatedParcels` field as well.